PROTOCOL.md

```diff
@@ -18,52 +21,54 @@ export interface OperationMessage {
 #### GQL_CONNECTION_INIT
 Client sends this message after plain websocket connection to start the communication with the server
 
-This message also triggers the `onConnect` callback on the server.
-
 - `payload: Object` : optional parameters that the client specifies in `connectionParams`
 
-#### GQL_CONNECTION_TERMINATE
-Client sends this message to terminate the connection.
-
 #### GQL_START
 Client sends this message to execute GraphQL operation
 - `id: string` : The id of the GraphQL operation to start
 - `payload: Object`:
     * `query: string` : GraphQL operation as string or parsed GraphQL document node
     * `variables?: Object` : Object with GraphQL variables
     * `operationName?: string` : GraphQL operation name
-    * `context?: any` : Execution context for the operation
     
 #### GQL_STOP
 Client sends this message in order to stop a running GraphQL operation execution (for example: unsubscribe)
 - `id: string` : operation id
     
+#### GQL_CONNECTION_TERMINATE
+Client sends this message to terminate the connection.    
+    
 ### Server -> Client
 
 #### GQL_CONNECTION_ERROR
-The server sends this message if `onConnect` callback returns `false` or throws an exception, and after sending this message, the server disconnects the client.
+The server may responses with this message to the `GQL_CONNECTION_INIT` from client, indicates the server rejected the connection.
 
 It server also respond with this message in case of a parsing errors of the message (which does not disconnect the client, just ignore the message).
```

Comment:
let's discuess about this behavior, i think if error occured on init state then we should just close the connection and let the client reset it's own state..